lib/client/ipam_block_reader_writer.go

```diff
@@ -82,10 +82,9 @@ func (rw blockReaderWriter) claimNewAffineBlock(
 		for _, pool := range givenPools {
 			// Validate the given pool is actually configured and matches the version.
 			if err := rw.checkPoolAvailability(pool); err != nil {
```

Comment:
checkPoolAvailability() performs a List of the pools which means it'll end up querying for each pool that is requested.   How easy would it be to make it query the pools once?